src/Concretize.hs

```diff
@@ -671,19 +671,33 @@ concretizeDefinition allowAmbiguity typeEnv globalEnv visitedDefinitions definit
           Left $ CannotConcretize definition
 
 -- | Find ALL functions with a certain name, matching a type signature.
-allFunctionsWithNameAndSignature :: Env -> String -> Ty -> [(Env, Binder)]
-allFunctionsWithNameAndSignature env functionName functionType =
-  filter (predicate . xobjTy . binderXObj . snd) (multiLookupEverywhere functionName env)
+allFunctionsWithNameAndSignature :: TypeEnv -> Env -> String -> Ty -> [(Env, Binder)]
```

Comment:
Is this name still accurate?

Reply:
Agree that is seems a little weird now with this fix...

Reply:
Good point! What do we think of simplifying to `allImplementations` where an implementation is meant to indicate a function that:

1. has a matching type signature.
2. has a matching name, either through membership in the set of paths when `name` denotes an interface, otherwise an exact match.

As an alternative, we could be a bit verbose but explicit and call it `allFunctionsWithNameAndSignatureOrAllImplementations`

Reply:
`allImplementations` 👍

Reply:
renamed in 3f7c96d!